### ISSUE TITLE
Incremental backups

### ISSUE THREAD

**Issue by dag** [openSUSE/snapper#368]
So I realize snapper isn't meant to be a backup tool, but there's a lot of overlap and it already has a bunch of infrastructure in place that would be needed for such a tool. I've been looking at existing solutions for btrfs incremental backups and I keep thinking it would be much easier if I could just do it with snapper, which I already know and have already set up. Would you consider supporting a few features to make it easier to use snapper also for backups? I'm imagining something like this (straw man proposal):

```bash
# create a new backup snapshot
# (optionally set --userdata backedup=no explicitly)
snapper create -c backup

# find the most recent backup on DEST,
# and iteratively send all the more recent backups using btrfs send -p
# then mark any sent snapshots with --userdata backedup=yes
snapper send-backups DEST

# variant that works over SSH
# figures out most recent backup based on backedup userdata
snapper send-backups | snapper receive-backups DEST

# delete any backedup=yes backups exceeding BACKUP_LIMIT_BACKEDUP,
# but always preserve the most recently backed up snapshot
# since it is needed for the next send-backups
# also delete any backedup=no backups exceeding BACKUP_LIMIT
snapper cleanup backup
```

* `DEST` could also be a configuration variable (under a different name), with a systemd timer creating and sending backups daily
* `BACKUP_LIMIT_BACKEDUP` could default to zero because you don't need to keep backed up snapshots around and you're better of using timeline if you want local snapshots for keeps
* `BACKUP_LIMIT` defaults to non-zero (say, 10), so you can create multiple backup "checkpoints" between sends
* Maybe using userdata for `backedup` is wrong if it isn't meant to be set by the user, but I don't know how easy it would be to add a new field to snapshots so thinking in terms of the userdata field was natural to me. Alternatively, it's a good thing because the user might want to use the backup cleanup logic without using snapper send-backups.

**Comment by NicoHood** [comment]
Try this:
https://github.com/wesbarnett/snap-sync

**Comment by rzerres** [comment]
Or this:
proposed to merge with snap-sync. Thus reached a working status to handle remote snapper ...
https://github.com/rzerres/snap-sync/wip-dash-v2

**Comment by rickysarraf** [comment]
A backup command in `snapper` is desirable. `snapper` has the ability to, if set proper privileges, not require `root` privileges. That is an important feature. And having all functionalities together in a single tool would be more appropriate.

**Comment by jwillikers** [comment]
I have to agree with @rickysarraf that it would be great to have users be able to backup snapshots without requiring `root` privileges using Snapper.

**Comment by marccollin** [comment]
+1 be able to save backup elsewhere is crucial

**Comment by Heavygrass** [comment]
+1
I'd like to request snapshot sync to other drives also (e.g. automatic send/receive).
I tried 
https://github.com/baod-rate/snap-sync - it is clearly abandoned and does NOT work at all (encountered multiple issues).
Without a possibility to sync to other drives, if a drive fails - all data is gone.
Thank you!

**Comment by aschnell** [comment]
Lets say yes, although root permissions are needed. Ideas to improve snbk can be reported as new issues.
